### PR TITLE
[Frontend] Add Mathjax support to html

### PR DESCRIPTION
**Pull request by Suryansh5545**
Added Mathjax support, the mathjax library is pulled from the cdn and a directive name mathjaxBind is used in favour of ng-bind-html. Added a small hint too for users to know how to use mathjax in trix editor

image

image

### CODE REVIEW

File: frontend/base.html [hunk 1]
```diff
@@ -34,6 +34,15 @@
...
     </script>
+    <script type="text/x-mathjax-config">
+        MathJax.Hub.Config({
+            tex2jax: {
+                inlineMath: [['$','$'], ['\\(','\\)']],
+                processEscapes: true
+                }
+        });
+    </script>
+    <script type="text/javascript" src="//cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
```
Comment (gchhablani) [inline frontend/base.html:45]: Is this change needed after MathJax bind?

File: frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html [hunk 2]
```diff
@@ -21,7 +21,8 @@
                                      trix-attachment-remove="trixAttachmentRemove(e, editor);"
                                      prevent-trix-file-accept = "true"
                                      placeholder="Evaluation Criteria">
-                          </trix-editor>
+                          <trix-editor>
+                        <code>Hint: To add Mathjax use $$....mathjax code...$$</code>
```
Comment (gchhablani) [inline frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:25]: What does this look like?

Comment (gchhablani) [reply frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:25]: Make this "To use MathJax in editor, use $$ <your mathjax code> $$". Use escape characters if needed.

Comment (gchhablani) [reply frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:25]: Make the same change everywhere.

File: frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html [hunk 3]
```diff
@@ -21,7 +21,8 @@
                                      trix-attachment-remove="trixAttachmentRemove(e, editor);"
                                      prevent-trix-file-accept = "true"
                                      placeholder="Evaluation Criteria">
-                          </trix-editor>
+                          <trix-editor>
```
Comment (gchhablani) [inline frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:24]: Why do you remove the closing </trix-editor> and start a new one?

Comment (gchhablani) [reply frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:24]: There is no closing tag for either of them.

File: frontend/src/views/web/challenge/evaluation.html [hunk 4]
```diff
@@ -29,7 +29,7 @@ <h5><a ng-if="challenge.isChallengeHost" class="pointer theme-text fs-16"
         </div>
         <div class="row">
             <div class="col s12">
-                <div ng-bind-html="challenge.page.evaluation_details" class="fs-16 w-300 evaluation-details trix-container"></div>
+                <div mathjax-bind="challenge.page.evaluation_details" class="fs-16 w-300 evaluation-details trix-container"></div>
```
Comment (gchhablani) [inline frontend/src/views/web/challenge/evaluation.html:32]: Does `mathjax-bind` entail everything that `ng-bind-html` had?

**Review by gchhablani — COMMENTED** [review]
Added some comments and questions.

File: frontend/base.html (hunk 1 above)
Comment (Suryansh5545) [reply frontend/base.html:45]: Yes This defines the structure of the command which mathjax recognize and the script pulls the Mathjax from the cdn

File: frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html (hunk 3 above)
Comment (Suryansh5545) [reply frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:24]: Typo

File: frontend/src/views/web/challenge/evaluation.html (hunk 4 above)
Comment (Suryansh5545) [reply frontend/src/views/web/challenge/evaluation.html:32]: Yes, as far as i saw using all the possible option in trix except insert image which is broken in trix and not a mathjax-bind issue

File: frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html [hunk 5]
```diff
@@ -21,7 +21,8 @@
                                      trix-attachment-remove="trixAttachmentRemove(e, editor);"
                                      prevent-trix-file-accept = "true"
                                      placeholder="Evaluation Criteria">
-                          </trix-editor>
+                      </trix-editor>
```
Comment (gchhablani) [inline frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html:24]: Are you sure the formatting (change in indent) is right?

Comment (Suryansh5545) [reply frontend/src/views/web/challenge/edit-challenge/edit-challenge-evaluation-criteria.html]: it was extra indented , pushed a fix
